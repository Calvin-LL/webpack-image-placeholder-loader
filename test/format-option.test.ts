import webpack from "webpack";

import compile from "./helpers/compile";
import execute from "./helpers/execute";
import getCompiler from "./helpers/getCompiler";
import readAsset from "./helpers/readAsset";

describe('"format" option', () => {
  it("should work without value", async () => {
    const compiler = getCompiler({});
    const stats = await compile(compiler);

    expect(
      execute(readAsset("main.bundle.js", compiler, stats as webpack.Stats))
    ).toMatchSnapshot("result");
  });

  it('should work with "base64" value', async () => {
    const compiler = getCompiler({
      format: "base64",
    });
    const stats = await compile(compiler);

    expect(
      execute(readAsset("main.bundle.js", compiler, stats as webpack.Stats))
    ).toMatchSnapshot("result");
  });

  it('should work with "hex" value', async () => {
    const compiler = getCompiler({
      format: "hex",
    });
    const stats = await compile(compiler);

    expect(
      execute(readAsset("main.bundle.js", compiler, stats as webpack.Stats))
    ).toMatchSnapshot("result");
  });

  it('should work with "rgb" value', async () => {
    const compiler = getCompiler({
      format: "rgb",
    });
    const stats = await compile(compiler);

    expect(
      execute(readAsset("main.bundle.js", compiler, stats as webpack.Stats))
    ).toMatchSnapshot("result");
  });

  it('should work with "array" value', async () => {
    const compiler = getCompiler({
      format: "array",
    });
    const stats = await compile(compiler);

    expect(
      execute(readAsset("main.bundle.js", compiler, stats as webpack.Stats))
    ).toMatchSnapshot("result");
  });
});
