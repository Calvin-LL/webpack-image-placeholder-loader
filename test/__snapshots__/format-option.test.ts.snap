// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`"format" option should work with "array" value: result 1`] = `
Array [
  107,
  117,
  72,
]
`;

exports[`"format" option should work with "base64" value: result 1`] = `"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAIAAACQd1PeAAAACXBIWXMAAAPoAAAD6AG1e1JrAAAADElEQVQI12PILvUAAAJ3ASkpvwmDAAAAAElFTkSuQmCC"`;

exports[`"format" option should work with "hex" value: result 1`] = `"#6b7548"`;

exports[`"format" option should work with "rgb" value: result 1`] = `"rgb(107, 117, 72)"`;

exports[`"format" option should work without value: result 1`] = `"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAIAAACQd1PeAAAACXBIWXMAAAPoAAAD6AG1e1JrAAAADElEQVQI12PILvUAAAJ3ASkpvwmDAAAAAElFTkSuQmCC"`;
